### PR TITLE
add SPLUNK_PROFILER_MEMORY_ENABLED

### PR DESCRIPTION
**Pull request by Kielek**
For now, splunk-otel-java is using
for CPU profiling
 - `splunk.profiler.enabled` - command switch
 - `SPLUNK_PROFILER_ENABLED` - env. var.

for memory profiling only command switch `splunk.profiler.memory.enabled`.

SignalFx for .NET is using only `SIGNALFX_PROFILER_ENABLED` for CPU profiling.
splunk-otel-js using only `SPLUNK_PROFILER_ENABLED` for CPU profiling.

We are in progress of adding memory profiling do SFx for .NET. It will be good to introduce common way to enable it (with different prefix).

Additional tasks:
 - for splunk-otel-java - add support to new env. variable
 - SFx for .NET - start using new env. when needed
 - splunk-otel-js - start using new env. when needed

### CODE REVIEW

**Comment by Kielek** [comment]
@mateuszrzeszutek, @breedx-splk, @seemk, @lachmatt, @pjanotti, @pellared, please check

File: specification/configuration.md [hunk 1]
```diff
@@ -191,7 +192,9 @@ instance using the following environment variables:
   precedence.
 - [4]: If stitching of RUM spans and APM spans is desired then this parameter
   MUST be set to `true`.
-- [5]: Applies only to instrumentation libraries with profiling capabilities. 
+- [5]: Applies only to instrumentation libraries with CPU profiling capabilities.
+  Profiling is experimental status.
```
Comment (MrAlias) [inline specification/configuration.md:196]: This will be stale when the profiling status changes.

```suggestion
```

**Review by MrAlias — APPROVED** [review]
Looks good overall :+1:

File: specification/configuration.md [hunk 2]
```diff
@@ -191,7 +192,9 @@ instance using the following environment variables:
...
+- [6]: Applies only to instrumentation libraries with memory profiling capabilities.
   Profiling is experimental status.
```
Comment (Kielek) [inline specification/configuration.md:198]:
```suggestion
```

**Pull request merged**